MoreLinq.Test/FullGroupJoinTest.cs

```diff
@@ -102,29 +108,39 @@ public void FullGroupPreservesOrder()
                 (1, 4),
                 (3, 3),
             };
-            var listB = new[] {
+            var listB = new[]
+            {
                 (4, 1),
                 (3, 1),
                 (2, 1),
                 (0, 1),
                 (3, 0),
             };
 
-            var result = listA.FullGroupJoin(listB, x => x.Item1, x => x.Item1, (key, first, second) => new { key, first, second }).ToList();
+            var result = FullGroupJoin(overloadCase, listA, listB, x => x.Item1).ToList();
 
             // Order of keys is preserved
-            result.Select(x => x.key).AssertSequenceEqual(3, 1, 2, 4, 0);
+            result.Select(x => x.Key).AssertSequenceEqual(3, 1, 2, 4, 0);
 
             // Order of joined elements is preserved
-            foreach (var res in result) {
-                res.first.AssertSequenceEqual(listA.Where(t => t.Item1 == res.key).ToArray());
-                res.second.AssertSequenceEqual(listB.Where(t => t.Item1 == res.key).ToArray());
+            foreach (var res in result)
+            {
+                res.First.AssertSequenceEqual(listA.Where(t => t.Item1 == res.Key).ToArray());
+                res.Second.AssertSequenceEqual(listB.Where(t => t.Item1 == res.Key).ToArray());
             }
         }
 
-        static T1 DummySelector<T1, T2, T3>(T1 t1, T2 t2, T3 t3)
+        private static IEnumerable<(int Key, IEnumerable<T> First, IEnumerable<T> Second)> FullGroupJoin<T>(OverloadCase overloadCase, IEnumerable<T> listA, IEnumerable<T> listB, Func<T, int> getKey)
         {
-            return t1;
+            switch (overloadCase)
+            {
+                case OverloadCase.ResultSelector:
+                    return listA.FullGroupJoin(listB, getKey, getKey, (k, f, s) => (k, f, s));
```

Comment:
Do you prefer spelling out `(k, f, s) => (k, f, s)` over just using the `ValueTuple.Create` as method group?